clang/lib/Sema/SemaChecking.cpp

```diff
@@ -1907,6 +1957,68 @@ Sema::CheckBuiltinFunctionCall(FunctionDecl *FDecl, unsigned BuiltinID,
       return ExprError();
     }
     break;
+
+  // We allow using CHERI builtins with any pointer type and (u)intcap_t.
+  // Note: The following are not polymorphic yet (and should be used so rarely
+  // that it probably doesn't matter):
+  // TODO: __builtin_cheri_cap_build
+  // TODO: __builtin_cheri_cap_type_copy
+  // TODO: __builtin_cheri_conditional_seal
+  // TODO: __builtin_cheri_cap_from_pointer
+  // TODO: __builtin_cheri_cap_to_pointer
+  case Builtin::BI__builtin_cheri_bounds_set:
+  case Builtin::BI__builtin_cheri_bounds_set_exact:
+  case Builtin::BI__builtin_cheri_perms_and:
+  case Builtin::BI__builtin_cheri_flags_set:
+  case Builtin::BI__builtin_cheri_offset_increment:
+  case Builtin::BI__builtin_cheri_offset_set:
+  case Builtin::BI__builtin_cheri_address_set:
+    if (SemaBuiltinCheriSet(*this, TheCall))
+      return ExprError();
+    break;
+  case Builtin::BI__builtin_cheri_seal:
+  case Builtin::BI__builtin_cheri_unseal:
+    // These work in the same way as the setters: value to be unsealed/sealed
+    // comes first and should therefore be the result type, second argument
+    // is type checked normally.
```

Comment:
1. If they're the same, why duplicate the body rather than just put the comment above the two case statements.

2. I think we do want to make them polymorphic in their second argument, which makes them different instead.